nvme-print-json.c

```diff
@@ -2854,3 +2856,27 @@ void json_output_status(int status)
 
 	json_output_object(root);
 }
+
+void json_output_error(const char *msg, va_list ap)
+{
+	struct json_object *root = json_create_object();
+	int len = ERROR_MSG_LEN;
+	char *error = (char *)malloc(ERROR_MSG_LEN);
+
+	if (!error)
+		return;
+
+	len = vsnprintf(error, len, msg, ap) + 1;
+	if (len > ERROR_MSG_LEN) {
+		error = (char *)realloc(error, len);
+		if (!error)
+			return;
+		vsnprintf(error, len, msg, ap);
+	}
```

Comment:
We are already using `asprintf` in the code base. So this could just use it as well. Avoids the open coded realloc etc.